docs: document lack of config file names (#17442)
eslint/eslint: docs/src/use/configure/configuration-files-new.md

})();
```

::: warning

> 
> Unlike other tools in the JavaScript ecosystem, ESLint does not automatically look for a config file with a name of `eslint.config.mjs` or `eslint.config.cjs`. The ESLint team [does not want to support these file extensions](https://github.com/eslint/eslint/issues/16580) because we want to keep things simple: users are expected to use the same format for their ESLint files that the rest of their project uses, as defined in the `type` field of the project's `package.json` file.
> 
> If you want to force a different kind of ESLint config, then name your config file accordingly and use the `-c` or `--config` flag when running ESLint. For example, if you want to have an ESM ESLint config in a CommonJS project, name the config file `eslint.config.mjs` and then run ESLint with:
> 
> ```shell
> ESLINT_USE_FLAT_CONFIG=true npx eslint --config eslint.config.mjs **/*.js
> ```
> 

Reviewer: This is a bit aggressive, plus it completely breaks the layout. I'm fine with adding a warning but let's keep it simple.
```suggestion
ESLint only automatically looks for a config file named `eslint.config.js` and does not look for `eslint.config.cjs` or `eslint.config.mjs`. If you'd like to specify a different config filename than the default, use the `--config` command line option.
```

Author: lgtm

You can prevent this search for `eslint.config.js` by setting the `ESLINT_USE_FLAT_CONFIG` environment variable to `true` and using the `-c` or `--config` option on the command line to specify an alternate configuration file, such as:

```shell

> ESLINT_USE_FLAT_CONFIG=true npx eslint --config some-other-file.js **/*.js

Author: While sort of unrelated to the warning that I added elsewhere, I think that using the full flag here makes the docs more clear.
